src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +42,10 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
-
-                this.beginRequestTimestamp.Value = timestamp;
-                this.client.Context.Operation.Id = httpContext.TraceIdentifier;
+                var requestTelemetry = new RequestTelemetry();
+                requestTelemetry.Context.Operation.Id = httpContext.TraceIdentifier;
```

Comment:
It should be `requestTelemetry.Id = httpContext.TraceIdentifier;` This will fix https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/340 Also consider fixing https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/333 in this PR - I'd appreciate it!